scaaml/io/dataset.py

```diff
@@ -675,41 +675,41 @@ def _check_shard_metadata(shard_info: Dict, dataset_path: Path) -> None:
         """
         # Check that only expected keys are present:
         si_keys = {
-            'examples',  # Checked by Dataset._check_metadata
-            'sha256',  # Checked by Dataset._check_sha256sums
-            'path',  # Checked by Dataset._check_sha256sums
-            'group',  # Checked against path
-            'key',  # Checked against path
-            'part',  # Checked against path
-            'size',  # Checked here
-            'chip_id',  # Checked that it is a non-negative integer
+            "examples",  # Checked by Dataset._check_metadata
+            "sha256",  # Checked by Dataset._check_sha256sums
+            "path",  # Checked by Dataset._check_sha256sums
+            "group",  # Checked against path
+            "key",  # Checked against path
+            "part",  # Checked against path
+            "size",  # Checked here
+            "chip_id",  # Checked that it is a non-negative integer
         }
         if set(shard_info.keys()) != si_keys:
-            raise ValueError(f'Shard info keys are: {shard_info.keys()} '
-                             f'expected: {si_keys}, in shard: {shard_info}')
+            raise ValueError(f"Shard info keys are: {shard_info.keys()} "
+                             f"expected: {si_keys}, in shard: {shard_info}")
         # Check that the info corresponds to the filename:
-        file_info = Dataset._shard_info_from_name(shard_info['path'])
-        for key in ['group', 'part']:
-            if file_info['shard_' + key] != shard_info[key]:
-                raise ValueError(f'{key} does not match filename, expected: '
-                                 f'{file_info["shard_" + key]}, got: '
-                                 f'{shard_info[key]}, in shard: {shard_info}')
+        file_info = Dataset._shard_info_from_name(shard_info["path"])
+        for key in ["group", "part"]:
+            if file_info["shard_" + key] != shard_info[key]:
```

Comment:
nit: slightly more readable by using an f-string I find:
```python
if file_info[f"shard_{key}"] != shard_info[key]:
```
And as it's needed for the exception, might be worth having a temp variable for it